cmake/modules/extensions.cmake

```diff
@@ -853,6 +853,39 @@ function(board_set_rimage_target target)
   zephyr_check_cache(RIMAGE_TARGET)
 endfunction()
 
+function(board_emu_args emu)
+  string(MAKE_C_IDENTIFIER ${emu} emu_id)
+  # Note the "_EXPLICIT_" here, and see below.
+  set_property(GLOBAL APPEND PROPERTY BOARD_EMU_ARGS_EXPLICIT_${emu_id} ${ARGN})
+endfunction()
+
+function(board_finalize_emu_args emu)
+  # If the application provided a macro to add additional emu
+  # arguments, handle them.
+  if(COMMAND app_set_emu_args)
+    app_set_emu_args()
+  endif()
```

Comment:
approved as it follows the same structure of the runners arg, so the following comment is mostly for future references if someone wonders about this construct.

I would normally be careful on this kind of construct as it requires a certain order of defining the function before calling `find_package(Zephyr)`, which seems to dates back from the boilerplate days and the former presence of `set_conf_files()` (now removed), see this commit message: https://github.com/zephyrproject-rtos/zephyr/commit/70a40813dff2c70dd1ce1a7e4b77096c93a428ce and instead consider a different pattern, but as this follows the structure of runners handling, then it's approved.